pkg/common/namespace_selection.go

```diff
@@ -103,3 +116,177 @@ func GetAllNamespaces(client kubernetes.Interface, labelSelector metav1.LabelSel
 
 	return namespacesNames, nil
 }
+
+type NamespaceSelectorReconciler struct {
+	Client     client.Client
+	selections map[string]NamespaceSelection
+	lock       sync.RWMutex
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *NamespaceSelectorReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	r.selections = make(map[string]NamespaceSelection)
+
+	neverEnqueue := predicate.NewPredicateFuncs(func(o client.Object) bool { return false })
+	mapToSingleton := func(_ client.Object) []reconcile.Request {
```

Comment:
This is a nice idea to reduce reconciles, however, a comment would be appreciated explaining this.